docs/guides/file-system-based-automated-bundle-generation.md

```diff
@@ -0,0 +1,133 @@
+# File-System-Based Automated Bundle Generation
+
+To use the automated bundle generation feature introduced in React on Rails v14.0.0, Please upgrade to use [Shakapacker v6.5.0](https://github.com/shakacode/shakapacker/tree/v6.5.0) at least. If you are currently using webpacker, please follow the migration steps available [here](https://github.com/shakacode/shakapacker/blob/master/docs/v6_upgrade.md).
+
+## Configuration
+
+### Enable nested_entries for Shakapacker
+To use the automated bundle generation feature, in the `webpacker.yml` file, set 
+
+```yml
+default:
+  ...
+  nested_entries: true
+```
+
+For more details, see [Configuration and Code ](https://github.com/shakacode/shakapacker#configuration-and-code) section in [shakapacker](https://github.com/shakacode/shakapacker/).
+
+### Configure Components Directory
+`components_directory` is the directory used to automatically detect and register components for usage on the rails view.
+Configure `config/initializers/react_on_rails`
+to set the name for `components_directory`. 
+
+```rb
+config.components_directory = "ror_components"
+```
+
+Now all React components inside `config.components_directory` will automatically be registered for the usage with `react_component` and `react_component_hash` helper methods provided by React on Rails.
+
+### Configure `auto_load_bundle` Option
+
+For automated component registry, `react_component` and `react_component_hash` view helper method tries to load generated bundle for component from the generated directory automatically per `auto_load_bundle` option. `auto_load_bundle` option in `config/initializers/react_on_rails` configures the default value that will be passed to component helpers. The default is `false`, and the option can be passed at the time of individual usage too.
+
+*Note: Starting from React on Rails version 15.0.0, the default value for `auto_load_bundle` option will be `true`.*
+
+You can change the value in `config/initializers/react_on_rails` by updating as follows:
+
+```rb
+config.auto_load_bundle = true
+```
+
+### Update `.gitignore` file
+React on Rails, per the settings in your `config/webpacker.yml`, automatically generates pack files for components to be registered in the `packs/generated` directory. To avoid committing generated files into the version control system, Please update `.gitignore` to have 
```

Comment:
1. Omit "per the settings in your `config/webpacker.yml`"? Not sure why we need it. If you mean the directory depend on it, the note below says it already. Or better, specify exactly how it depends.
2. "Please" shouldn't be capitalized.